plutus-core/plutus-ir/src/PlutusIR/Compiler.hs

```diff
@@ -95,7 +97,7 @@ availablePasses :: [Pass uni fun]
 availablePasses =
     [ Pass "unwrap cancel"        (onOption coDoSimplifierUnwrapCancel)       (pure . Unwrap.unwrapCancel)
     , Pass "beta"                 (onOption coDoSimplifierBeta)               (pure . Beta.beta)
-    , Pass "inline"               (onOption coDoSimplifierInline)             Inline.inline
+    , Pass "inline"               (onOption coDoSimplifierInline)             (\t -> do { hints <- asks (view (ccOpts . coInlineHints)); Inline.inline hints t })
```

Comment:
Why is `asks` redundant?

I think it's nicer to keep the general pattern of "config args come first, then the 'main' argument", the glue code isn't so bad.

Reply:
> Why is `asks` redundant?
> 
`asks view` === `view` because view works over monadreader 
```
view :: MonadReader s m => Getting a s a -> m a
```

Reply:
Wow, I hate it.

Reply:
lol